crates/title_bar/src/application_menu.rs

```diff
@@ -1,145 +1,349 @@
-use ui::{prelude::*, ContextMenu, NumericStepper, PopoverMenu, PopoverMenuHandle, Tooltip};
+use gpui::View;
+use smallvec::SmallVec;
+use ui::{prelude::*, ContextMenu, PopoverMenu, PopoverMenuHandle, Tooltip};
+
+use install_cli;
+use terminal_view::terminal_panel;
+use workspace;
+
+#[derive(Copy, Clone)]
+enum MenuType {
+    Zed,
+    File,
+    Edit,
+    Selection,
+    View,
+    Go,
+    Window,
+    Help,
+}
+
+#[derive(Clone)]
+struct MenuItem {
+    menu_type: MenuType,
+    handle: PopoverMenuHandle<ContextMenu>,
+}
+
+impl MenuItem {
+    fn id(&self) -> &'static str {
+        match self.menu_type {
```

Comment:
I think it would be preferable to use the `AppContext::get_menus()` API to fill out this data, rather than duplicating our menu definitions across two files. I don't think we use submenus very often, but where we do we could simply flatten them into their containing menus :)